service/history/replicationTaskExecutor.go

```diff
@@ -256,3 +275,33 @@ FilterLoop:
 	}
 	return shouldProcessTask, nil
 }
+
+func (e *replicationTaskExecutorImpl) cleanupWorkflowExecution(ctx context.Context, namespaceID string, workflowID string, runID string) (retErr error) {
+	nsID := namespace.ID(namespaceID)
+	ex := commonpb.WorkflowExecution{
+		WorkflowId: workflowID,
+		RunId:      runID,
+	}
+	wfCtx, releaseFn, err := e.workflowCache.GetOrCreateWorkflowExecution(ctx, nsID, ex, workflow.CallerTypeTask)
+	if err != nil {
+		return err
+	}
+	defer func() { releaseFn(retErr) }()
+	mutableState, err := wfCtx.LoadWorkflowExecution(ctx)
+	if err != nil {
+		return err
+	}
+	lastWriteVersion, err := mutableState.GetLastWriteVersion()
+	if err != nil {
+		return err
+	}
+	return e.deleteManager.DeleteWorkflowExecution(
```

Comment:
Synced offline. The implementation of `DeleteWorkflowExecution` will get workflow close event (for deleting vis record), which does not exist for running workflows. 

Need some refactoring work around deleting visibility records to make this work. Blocking this PR for now.

Reply:
Pending https://github.com/temporalio/temporal/pull/2636

Reply:
👍 LGTM once #2636 is landed.